### ISSUE TITLE
VScode: Refresh should also reset animation-tickI()

### ISSUE THREAD

**Issue by NigelBreslaw** [slint-ui/slint#6541]
Working on tweaking animations that use animation-tick() is painful as the refresh button doesn't reset the value from animation-tick(). Instead I had to restart the LSP which also kills the live preview and needs it to be re-opened.

**Comment by hunger** [comment]
I think this is due to the animation tick being taken from the "outside" UI and just forwarded to the inside UI as is. We need to have some kind of mapping in the interpreter for that.

**Comment by ogoffart** [comment]
We'd need an (private) API to reset the INITIAL_INSTANT
But that would also reset all the timers used in the preview UI itself, which may not be a good idea.
We could imagine to put that in the SlintContext, but even then, we probably need the same SlintContext for the UI and its preview (or do we?)

We could also make a hack in the interpreter to have a different offset for animation-tick for a given component instance.

Anyway, this is not a trivial issue, and i'm not sure this is worth it. The animation-tick should be used as a relative way to make animations. and not as an absolute value.

**Comment by NigelBreslaw** [comment]
I think reseting all the timers is the expected behaviour? The refresh is basically restart?

**Comment by hunger** [comment]
I agree with Nigel, but technically that will be tricky to do though: The entire thing is *one* UI that combines our UI and the interface you are previewing. We need to make it possible for differetn parts of the item tree to see different `animation-tick()`s for this to work.

**Comment by tronical** [comment]
> I think reseting all the timers is the expected behaviour? The refresh is basically restart?

`Timer` elements are reset.

But regarding, `animation-tick()`, are you storing the absolute value somewhere? Could you provide a small example? Maybe that gives an idea how to work around it in the preview.
